src/oscar/apps/basket/middleware.py

```diff
@@ -176,13 +180,14 @@ def get_basket(self, request):
         # Cache basket instance for the during of this request
         request._basket_cache = basket
 
-        if num_baskets_merged > 0:
+        if num_items_merged > 0:
+            # show warning only if items have been merged
             messages.add_message(
                 request,
                 messages.WARNING,
                 _(
-                    "We have merged a basket from a previous session. Its contents "
-                    "might have changed."
+                    f"We have merged {num_items_merged} items from a "
```

Comment:
Unfortunately f strings don't work in translatable strings yet, please use %(name)s like it's done in other places.